kano_updater/utils.py

```diff
@@ -677,3 +679,58 @@ def disable_power_button():
     except Exception:
         # Kano Peripherals doesn't support this function
         pass
+
+
+def verify_kit_is_plugged():
+    """
+    On Computer Kit 2 Pro, verify that the battery is plugged in and not depleting.
+    For now, we rely on the user to tell us.
+
+    Returns:
+        is_plugged - bool whether the kit is plugged in or not.
+    """
+    ck2_pro = False
+    is_battery_low = False
+    is_plugged = True
+
+    try:
+        from kano_peripherals.wrappers.detection import is_ck2_pro
+        ck2_pro = is_ck2_pro()
+
+        from kano_peripherals.ck2_pro_hat.driver.high_level import get_ck2_pro_hat_interface
+        ck2pro_iface = get_ck2_pro_hat_interface()
+        is_battery_low = (ck2pro_iface and ck2pro_iface.is_battery_low())
+    except:
+        # Kano Peripherals doesn't support this function
+        pass
```

Comment:
Should we also set something here - the hat may exist but if we can't access the interface then we are in trouble. I'm not sure whether we should assume that the battery is fine in this situation or say that it is too risky. Presumably we should warn anyway. What do you thing @radujipa